src/lava/magma/core/model/py/ports.py

```diff
@@ -108,50 +241,89 @@ def peek(self) -> ty.Tuple[int, int]:
+    (PyInPort) over a channel. PyOutPort can send (send()) the data, which
+    adds it to the channel or clear (flush()) the channel to remove any data of
+    the channel.
+    The different class attributes are used to select the type of OutPorts via
+    LavaPyType declarations in PyProcModels, e.g., LavaPyType(
+    PyOutPort.VEC_DENSE, np.int32, precision=24) creates a PyOutPort.
+    A PyOutPort (source) can be connected to one or multiple PyInPorts (target).
+
+    Class attributes
+    ----------------
+    VEC_DENSE : PyOutPortVectorDense, default=None
+        Type of PyOutPort.
+
+    VEC_SPARSE : PyOutPortVectorSparse, default=None
+        Type of PyOutPort.
+
+    SCALAR_DENSE : PyOutPortScalarDense, default=None
+        Type of PyOutPort.
+
+    SCALAR_SPARSE : PyOutPortScalarSparse, default=None
+        Type of PyOutPort.
+    """
 
     VEC_DENSE: ty.Type["PyOutPortVectorDense"] = None
     VEC_SPARSE: ty.Type["PyOutPortVectorSparse"] = None
     SCALAR_DENSE: ty.Type["PyOutPortScalarDense"] = None
     SCALAR_SPARSE: ty.Type["PyOutPortScalarSparse"] = None
 
-    def __init__(self, csp_send_ports: ty.List[CspSendPort], *args):
-        self._csp_send_ports = csp_send_ports
-        super().__init__(*args)
-
-    @property
-    def csp_ports(self) -> ty.List[AbstractCspPort]:
-        """Returns all csp ports of the port."""
-        return self._csp_send_ports
-
     @abstractmethod
     def send(self, data: ty.Union[np.ndarray, int]):
+        """Abstract method to send data to the connected Port PyInPort (target).
+
+        Parameters
+        ----------
+        data : ndarray or int
+            The data (vector or scalar) to be sent to the PyInPort (target).
+        """
         pass
 
     def flush(self):
+        """TBD"""
         pass
 
 
 class PyOutPortVectorDense(PyOutPort):
+    """Python implementation of PyOutPort for dense vector data."""
+
     def send(self, data: np.ndarray):
-        """Sends data only if port is not dangling."""
-        for csp_port in self._csp_send_ports:
+        """Abstract method to send data to the connected in Port (target).
+
+        Sends data only if Port is not dangling.
+
+        Parameters
+        ----------
+        data : ndarray
+            The data vector to be sent to the in Port (target).
+        """
+        for csp_port in self.csp_ports:
             csp_port.send(data)
 
 
 class PyOutPortVectorSparse(PyOutPort):
+    """Python implementation of PyOutPort for sparse vector data."""
     def send(self, data: np.ndarray, idx: np.ndarray):
+        """TBD"""
         pass
 
 
 class PyOutPortScalarDense(PyOutPort):
+    """Python implementation of PyOutPort for dense scalar data."""
     def send(self, data: int):
+        """TBD"""
         pass
 
 
 class PyOutPortScalarSparse(PyOutPort):
+    """Python implementation of PyOutPort for sparse scalar data."""
     def send(self, data: int, idx: int):
+        """TBD"""
         pass
 
 
```

Comment:
Is the code in line 330-333 actually correct? Why do we have these definitions outside any class?